### DISCUSSION TITLE
"Preset package is missing a renovate-config entry" using Github Package Registry

### DISCUSSION THREAD

**Discussion by rebeccatang16** [renovatebot/renovate#8266] [Request Help]
We are switching from npmjs to Github Package Registry for npm packages, and are now encountering this error from self-hosted renovate (version 1.2.0):
`Preset package is missing a renovate-config entry (@orgName)`

Our `@orgName/renovate-config` repo has a "renovate-config" section in its package.json, plus it worked before the registry change.

This continues the discussion started here https://github.com/whitesource/renovate-on-prem/issues/130.

**Comment by wyardley** [comment]
And, as mentioned there, it seems to be due to GH packages seemingly limiting metadata to certain fields.

See, e.g.,
https://github.community/t/npm-bin-field-is-not-exported-in-published-metadata/128402

**Reply by rarkins** [reply]
It seems we don't have much room to move unless GitHub makes a change to this or at least clarifies which fields are included and whether we can overload any.

**Comment by wyardley** [comment]
One workaround would be to actually grab the package and its `package.json`, but I'd totally understand if this doesn't seem worth it. We do have a ticket open to get more info on this / clarify.

Is there any config option we could use to trick Renovate into downloading the renovate-config package from NPM but have the `.npmrc` configured for GH packages otherwise?

**Reply by viceice** [reply]
Nope. but i would suggest you to add the shared config to a github repo and use `github>org/repo:file` as preset. `:file` can be removed if the preset filename is `default.json`

**Reply by rarkins** [reply]
I think they're trying to avoid needing to update many repos which already have `"extends": ["@orgname"]` in their `renovate.json`

**Reply by wyardley** [reply]
Yeah that’s true, tho it may end up being the way we have to go. We do have some non default presets, but we’re going to try forcing the override in the short term to try and get something working again.

**Comment by rebeccatang16** [comment]
Testing out using the `force` option like discussed in the previous thread using the cli, but can't seem to get it to work. The test `config.js` I'm providing is:
```
module.exports = {
  onboardingConfig: {
    extends: ['github>orgName/renovate-config']
  },
  repositories: ['orgName/renovate-config'],
  platform: 'github',
  token: 'XXXXXX',
  logLevel: 'debug',
  force: {
    extends: ['github>orgName/renovate-config']
  },
};
```

This gives me this error:
```
 INFO: Throwing preset error
       "validationError": "Cannot find preset's package (github>orgName/renovate-config)"
FATAL: Fatal error: config-validation
       "err": {
         "validationError": "Cannot find preset's package (github>orgName/renovate-config)",
         "message": "config-validation",
         "stack": "Error: config-validation\n    at resolveConfigPresets (/Users/rktang/.nvm/versions/node/v12.18.1/lib/node_modules/renovate/dist/config/presets/index.js:234:35)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)"
       }
(node:76372) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'redisUrl' of undefined
    at Object.cleanup (/Users/rktang/.nvm/versions/node/v12.18.1/lib/node_modules/renovate/dist/util/cache/package/index.js:68:16)
    at Object.globalFinalize (/Users/rktang/.nvm/versions/node/v12.18.1/lib/node_modules/renovate/dist/workers/global/initialize.js:71:18)
    at Object.start (/Users/rktang/.nvm/versions/node/v12.18.1/lib/node_modules/renovate/dist/workers/global/index.js:95:22)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:76372) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:76372) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

If I take out the `force` section, it appears to work as expected. What am I doing wrong?

**Reply by rarkins** [reply]
Is it on github.com or self-hosted GHE? If self-hosted then you need local> instead of github>

**Reply by wyardley** [reply]
It's cloud hosted GHE (so just regular github.com)

**Reply by rarkins** [reply]
Try running with LOG_LEVEL=debug in env as it gives additional config debugging

**Reply by rebeccatang16** [reply]
Additional debug output has lines like:
```
DEBUG: GitHub 404
       "url": "https://api.github.com/repos/orgName/renovate-config/contents/default.json"
DEBUG: Failed to retrieve default.json from repo
       "statusCode": 404,
       "url": "https://api.github.com/repos/orgName/renovate-config/contents/default.json"
DEBUG: default.json preset not found - trying renovate.json
...
```

I can curl for the default.json using the same token just fine. Also made sure `RENOVATE_TOKEN` was set to the same token in the env. I also tried setting `GITHUB_APP_KEY` and `GITHUB_APP_ID` in the env and taking out the token bit, but I get the same errors.

**Comment by wyardley** [comment]
ps - we do have a confirmation from GHE support that this is currently a limitation (as far as NPM packages go), so just in case this comes up for someone else. Frankly, it wasn't something we imagined would be an issue. I'm not sure if other artifact repositories that implement an NPM type interface also have a similar limitation or not.

> Our engineers have confirmed to me that dynamic fields in the package.json file are not supported by the GitHub Package Registry. The GitHub Package Registry only supports interacting with the NPM CLI, so only the fields required by the NPM CLI are stored.

> I will open an internal feature request and I'll add your feedback to it. I can’t give you an ETA on when or if this change will be implemented. but please keep an eye on our blog and community forums for news about this.

**Reply by viceice** [reply]
`github>orgName/renovate-config` preset must be public if used in bot config.

the token and hostRules are not yet available while parsing bot config. There should be an issue for this.

**Reply by rarkins** [reply]
If you have confidential settings you could have the public repo preset simply extend another private repo preset
